external/libYaml.cmake

```diff
@@ -0,0 +1,67 @@
+# -----------------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2019, Numenta, Inc.  Unless you have purchased from
+# Numenta, Inc. a separate commercial license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# -----------------------------------------------------------------------------
+# This downloads and builds the libyaml library.
+#
+# libyaml  - This is a SAX parser which means that it performs callbacks
+#            or returns tokens for each symbol it parses from the yaml text.  
+#            Therefore the interface (Value.cpp) must create the internal structure.
+#            This is a YAML 1.1 implementation.
+#
+#            The current release is 0.2.2
+#            The repository is at https://github.com/yaml/libyaml
+#            Documentation is at https://pyyaml.org/wiki/LibYAML
+#
+#            There is a problem in version 0.2.2; it does not build.
+#            Use the current master.
+#
+if(EXISTS   ${REPOSITORY_DIR}/build/ThirdParty/share/libyaml.zip)
+    set(URL ${REPOSITORY_DIR}/build/ThirdParty/share/libyaml.zip)
+elif(EXISTS ${REPOSITORY_DIR}/build/ThirdParty/share/libyaml.tar.gz)
+    set(URL ${REPOSITORY_DIR}/build/ThirdParty/share/libyaml.tar.gz)
+else()
+    set(URL "https://github.com/yaml/libyaml/archive/master.zip")
```

Comment:
please use a fixed release/tag/commit

Reply:
> please use a fixed release/tag/commit

The current release is 0..2.2 but it did not build.
so as mentioned in the comments, we have to use the master repository until they release again.

Reply:
> we have to use the master repository until they release again. 

yes, but use specific commit that works, so the "version" is fixed

Reply:
How do I do that?  I think I can only have access to whatever is the latest in the archives.